src/OpenSage.Game/Logic/Object/Update/AIUpdate/AIUpdate.cs

```diff
@@ -93,6 +93,20 @@ public class AIUpdate : UpdateModule
     /// </summary>
     public List<Vector3> TargetPoints { get; set; }
 
+    public float CurrentLocomotorSpeed
+    {
+        get
+        {
+            if (CurrentLocomotor != null)
+            {
+                return CurrentLocomotor.GetMaxSpeedForCondition(GameObject.BodyModule.DamageState);
+            }
+
+            Logger.Info("No current locomotor!");
```

Comment:
warn/error? Or is this fine? I guess I assume because it has an exclamation mark that it's not fine lol

Reply:
I actually copied the original in logging rather than asserting here. I don't know how carefully the original code decided between logging vs asserting, but for now I'm mostly just copying it.

Talking of which, I'd like to change the logging system so each class doesn't need its own logger instance. Building on #1343, maybe we should have a `DebugUtility.Log`.